Added: Email and Name validations

Ephoniz opened the pull request:
Added spec file for Authors and some validations on Author class.

Ephoniz committed:
Added: Email and Name validations

DocOnDev reviewed (changes requested):
Thank you for making your first pull request. A few things before I can merge this one:

1. Change the raised error(s) to an ArgumentError - https://ruby-doc.org/core-2.7.0/ArgumentError.html
2. Verify that the name provided is a String and raise an ArgumentError if it is not
3. Drop the to_s on the assignment to the instance variable

Ephoniz committed:
Corrected author validations and specs

Ephoniz commented:
I've applied the changes mentioned above: 

1. Changed raise StandardError to ArgumentError
2. Added validations + test for when name and email are not strings.
3. Dropped the to_s from author.
